lang/typescript/src/utils/regions.ts

```diff
@@ -46,3 +99,100 @@ export function getRegionConfig(countryCode: string): RegionYamlConfig | null {
 
   return null;
 }
+
+export function addressUsesScript(
+  fieldDefinition: FieldConcatenationRule[],
+  address: Partial<Address>,
+  script: Script,
+): boolean {
+  const charsetFound = fieldDefinition.reduce((matchesCharset, field) => {
+    if (address[field.key]) {
+      const value = address[field.key];
+      if (value) {
+        return matchesCharset || stringUsesScript(value, script);
+      }
+    }
+
+    return matchesCharset;
+  }, false);
+
+  return charsetFound;
+}
+
+export function getConcatenationRules(
+  config: RegionYamlConfig,
+  address: Address,
+  extendedField: keyof Address,
+): FieldConcatenationRule[] | undefined {
+  if (config.combined_address_format === undefined) {
+    return undefined;
+  }
+
+  const combinedAddressFormat = config.combined_address_format;
+  let concatenationRules = combinedAddressFormat[extendedField];
+
+  if (!isFieldConcatenationRuleArray(concatenationRules)) {
+    throw Error(
+      `combined_address_format.${extendedField} must be an array. It is: ${JSON.stringify(concatenationRules, null, 2)}`,
+    );
+  }
+
+  // Check for a valid language override and use the first one found
+  const langOverrides = Object.keys(langScripts).filter(
+    (lang) => lang in combinedAddressFormat,
+  );
+  for (const lang of langOverrides) {
+    const overrideConfig = config.combined_address_format[lang];
+    if (
+      isLanguageSpecificAddressFormat(overrideConfig) &&
+      isFieldConcatenationRuleArray(overrideConfig[extendedField]) &&
+      addressUsesScript(
+        overrideConfig[extendedField],
+        address,
+        langScripts[lang],
+      )
+    ) {
+      concatenationRules = overrideConfig[extendedField];
+      break;
+    }
+  }
```

Comment:
This is the important bit that basically does this:

1. Get all languages overrides defined on this config (in our Taiwan case we should find one, `zh-TW`)
2. For each override, call `addressUsesScript` and check if the Address has a matching character set for the lanuage specified in the override
   - Matching characters found? Use that override config instead (`zh-TW` config for our example)
   - No matching characters? Use the normal `combined_address_format`